kazoo/client.py

```diff
@@ -217,11 +217,17 @@ def __init__(self, hosts='127.0.0.1:2181',
             self._conn_retry = KazooRetry(**connection_retry)
         elif type(connection_retry) is KazooRetry:
             self._conn_retry = connection_retry
+        elif connection_retry is not None:
+            raise ConfigurationError("Unexpected connection_retry type %r"
+                                     % connection_retry)
```

Comment:
Or are u just following the existing pattern of configuration error (if so thats fine, just wondering...)